src/Build/Evaluation/Evaluator.cs

```diff
@@ -1927,17 +1927,45 @@ ProjectRootElement InnerCreate(string _, ProjectRootElementCacheBase __)
         private LoadImportsResult ExpandAndLoadImportsFromUnescapedImportExpression(string directoryOfImportingFile, ProjectImportElement importElement, string unescapedExpression,
                                             bool throwOnFileNotExistsError, out List<ProjectRootElement> imports)
         {
+            imports = new List<ProjectRootElement>();
+
+            bool ignoreInvalidImports = _loadSettings.HasFlag(ProjectLoadSettings.IgnoreInvalidImports);
```

Comment:
nit:
Common case should be !String.IsNullOrWhiteSpace(importExpressionEscaped), so move this under that if?

Reply:
Yes good point.  I was going to use the variable `ignoreInvalidImports` later in this method but didn't end up doing that.  I'll move this down inside the `if`.  I'll also do the bitwise comparison myself as I just read that `Enum.HasFlag()` is slow :(

https://stackoverflow.com/questions/7368652/what-is-it-that-makes-enum-hasflag-so-slow

Should I fix the other two calls to `HasFlag()` in the evaluator?  The first one isn't called that much but the second one is on the hot path

1. https://github.com/dotnet/msbuild/blob/master/src/Build/Evaluation/Evaluator.cs#L1768
2. https://github.com/dotnet/msbuild/blob/master/src/Build/Evaluation/Evaluator.cs#L2400

Or should we do that in a different change?

Reply:
Up to you—I'm happy with this change as-is, but the change is straightforward enough that it seems unlikely something would go wrong.

Reply:
I'm going to do a separate change to get rid of the `HasFlag()` calls, just to be safe.  Feel free to merge.  Thanks!